src/canisters/frontend/ic-certified-assets/src/state_machine.rs

```diff
@@ -96,15 +95,48 @@ pub struct State {
     authorized: Vec<Principal>,
 
     asset_hashes: AssetHashes,
+    redirect_enabled: bool,
+}
+
+impl Default for State {
+    fn default() -> Self {
+        Self {
+            assets: HashMap::default(),
+            chunks: HashMap::default(),
+            next_chunk_id: ChunkId::default(),
+            batches: HashMap::default(),
+            next_batch_id: BatchId::default(),
+            authorized: Vec::default(),
+            asset_hashes: AssetHashes::default(),
+            redirect_enabled: true,
+        }
+    }
 }
 
 #[derive(Clone, Debug, CandidType, Deserialize)]
 pub struct StableState {
     authorized: Vec<Principal>,
     stable_assets: HashMap<String, Asset>,
+    redirect_enabled: bool,
 }
 
 impl State {
+    fn get_asset(&self, key: &Key) -> Result<&Asset, String> {
+        self.assets
+            .get(key)
+            .or_else(|| {
+                if self.redirect_enabled {
+                    redirect(key)
+                        .into_iter()
+                        .find(|redirect_key| self.assets.contains_key(redirect_key))
+                        .and_then(|found_key| self.assets.get(&found_key))
```

Comment:
Does this work?
```suggestion
                        .find_map(|redirect_key| self.assets.get(&redirect_key))
```

Reply:
Yes, this works. Thanks for the suggestion, I wasn't aware of this function